optapy-core/src/main/python/test/__init__.py

```diff
@@ -0,0 +1,609 @@
+        """
+        ...
+
+    def rewards(self, times: int = None, message: str = None):
+        """
+        Asserts that the Constraint being tested, given a set of facts, results in a given number of rewards.
+
+        Ignores the constraint and match weights: it only asserts the number of matches
+        For example: if there are two matches with weight of 10 each, this assertion will check for 2 matches.
+
+        :param times: sometimes None, the expected number of rewards. If not provided, it raises an AssertionError
+                      when there are no rewards
+        :param message: sometimes None, description of the scenario being asserted
+
+        :raises AssertionError: when the expected reward is not observed if times is provided, or
+                                when there are no rewards if times is not provided
+
+        """
+        from java.lang import AssertionError as JavaAssertionError  # noqa
+        try:
+            if times is None and message is None:
+                self.delegate.rewards()
+            elif times is not None and message is None:
+                self.delegate.rewards(times)
+            elif times is None and message is not None:
+                self.delegate.rewards(message)
+            else:
+                self.delegate.rewards(times, message)
+        except JavaAssertionError as e:
+            raise AssertionError(e.getMessage())
+
+    @overload
+    def rewards_with(self, match_weight_total: int) -> None:
+        """
+        Asserts that the Constraint being tested, given a set of facts, results in a specific reward.
+
+        Ignores the constraint weight: it only asserts the match weights.
+        For example: a match with a match weight of 10 on a constraint with a constraint weight of -2hard reduces the
+        score by -20hard. In that case, this assertion checks for 10.
+
+        :param match_weight_total: the expected reward
+        :raises AssertionError: when the expected reward is not observed
+        """
+        ...
+
+    @overload
+    def rewards_with(self, match_weight_total: int, message: str) -> None:
+        """
+        Asserts that the Constraint being tested, given a set of facts, results in a specific reward.
+
+        Ignores the constraint weight: it only asserts the match weights.
+        For example: a match with a match weight of 10 on a constraint with a constraint weight of -2hard reduces the
+        score by -20hard. In that case, this assertion checks for 10.
+
+        :param match_weight_total: the expected reward
+        :param message: sometimes None, description of the scenario being asserted
+        :raises AssertionError: when the expected reward is not observed
+        """
+        ...
+
+    def rewards_with(self, match_weight_total: int, message: str = None):
+        return self.rewardsWith(match_weight_total, message)
+
+    @overload
+    def rewardsWith(self, match_weight_total: int) -> None:
```

Comment:
I don't understand. Why would we have both `rewards_with` and `rewardsWith`, the latter not being "the Python way"? This makes the API confusing.

Reply:
If we want `JImplements` to do API consistency checks, then we need `rewardsWith` (as `OptaPlanner` use `rewardsWith`, so `JImplements` will check for `rewardsWith`). The API being Pythonic (i.e. with `underscores`) was due to comments in this reddit thread (https://www.reddit.com/r/Python/comments/q2euxw/a_new_constraint_solver_in_python_for_the_vehicle/ ; https://github.com/optapy/optapy/issues/22).

Reply:
That is unfortunate - we want `rewards_with`, as that is the Python way. At the same time, I don't see why we would have `rewardsWith` as well, that is just confusing. So no `JImplements` for us. :-(

Please remove the `rewardsWith` methods from this API.